cloud/aws/elasticache/common/monitors-elasticache.tf

```diff
@@ -0,0 +1,174 @@
+		parameterized_subject = "[{{ruleSeverity}}]{{{detectorName}}} {{{ruleName}}} ({{inputs.signal.value}}) on {{{dimensions}}}"
+	}
+
+	rule {
+		description           = "is too high > ${var.swap_threshold_warning}"
+		severity              = "Warning"
+		detect_label          = "WARN"
+		disabled              = coalesce(var.swap_disabled_warning, var.swap_disabled, var.detectors_disabled)
+		notifications         = coalescelist(var.swap_notifications_warning, var.swap_notifications, var.notifications)
+		parameterized_subject = "[{{ruleSeverity}}]{{{detectorName}}} {{{ruleName}}} ({{inputs.signal.value}}) on {{{dimensions}}}"
+	}
+
+}
+
+resource "signalfx_detector" "free_memory" {
+	name = "${join("", formatlist("[%s]", var.prefixes))}[${var.environment}] ElastiCache free memory"
+
+	program_text = <<-EOF
+		signal = data('FreeableMemory', filter=filter('namespace', 'AWS/ElastiCache') and filter('stat', 'lower') and filter('CacheNodeId', '*') and ${module.filter-tags.filter_custom}).rateofchange()${var.free_memory_aggregation_function}.${var.free_memory_transformation_function}(over='${var.free_memory_transformation_window}')
+		detect(when(signal < ${var.free_memory_threshold_critical})).publish('CRIT')
+		detect(when(signal < ${var.free_memory_threshold_warning})).publish('WARN')
+	EOF
+
+	rule {
+		description           = "is too low < ${var.free_memory_threshold_critical}"
+		severity              = "Critical"
+		detect_label          = "CRIT"
+		disabled              = coalesce(var.free_memory_disabled_critical, var.free_memory_disabled, var.detectors_disabled)
+		notifications         = coalescelist(var.free_memory_notifications_critical, var.free_memory_notifications, var.notifications)
+		parameterized_subject = "[{{ruleSeverity}}]{{{detectorName}}} {{{ruleName}}} ({{inputs.signal.value}}) on {{{dimensions}}}"
+	}
+
+	rule {
+		description           = "is too low < ${var.free_memory_threshold_warning}"
+		severity              = "Warning"
+		detect_label          = "WARN"
+		disabled              = coalesce(var.free_memory_disabled_warning, var.free_memory_disabled, var.detectors_disabled)
+		notifications         = coalescelist(var.free_memory_notifications_warning, var.free_memory_notifications, var.notifications)
+		parameterized_subject = "[{{ruleSeverity}}]{{{detectorName}}} {{{ruleName}}} ({{inputs.signal.value}}) on {{{dimensions}}}"
+	}
+
+}
+
+resource "signalfx_detector" "evictions_growing" {
+	name = "${join("", formatlist("[%s]", var.prefixes))}[${var.environment}] ElastiCache evictions growing"
+
+	program_text = <<-EOF
+		A = data('Evictions', filter=filter('namespace', 'AWS/ElastiCache') and filter('stat', 'mean') and filter('CacheNodeId', '*') and ${module.filter-tags.filter_custom}).rateofchange()${var.evictions_growing_aggregation_function}.${var.evictions_growing_transformation_function}(over='${var.evictions_growing_transformation_window}')
+		B = (A).rateofchange()
+		signal = (B/A*100)
```

Comment:
is this useful to divide again while you use rateofchange() ? only using B is not enough ?

Reply:
will investtgate